osmoutils/accum/accum_test.go

```diff
@@ -1567,3 +1569,62 @@ func (suite *AccumTestSuite) TestGetTotalShares() {
 	suite.Require().Equal(expectedShares[0], accumOneShares)
 	suite.Require().Equal(expectedShares[1], accumTwoShares)
 }
+
+func (suite *AccumTestSuite) TestAddToUnclaimedRewards() {
+	// We setup store and accum
+	// once at beginning.
+	suite.SetupTest()
+
+	// Setup.
+	var (
+		accObject = accumPackage.MakeTestAccumulator(suite.store, testNameOne, initialCoinsDenomOne, emptyDec)
+	)
+
+	tests := map[string]struct {
+		positionName             string
+		unclaimedRewardsAddition sdk.DecCoins
+		expectedError            error
+	}{
+		"valid update": {
+			positionName:             validPositionName,
+			unclaimedRewardsAddition: initialCoinsDenomOne,
+		},
+		"error: negative addition": {
+			positionName:             validPositionName,
+			unclaimedRewardsAddition: negativeCoins,
+			expectedError:            accumPackage.NegativeRewardsAdditionError{AccumName: accObject.GetName(), PositionName: validPositionName},
+		},
+		"invalid position - different name": {
+			positionName:  invalidPositionName,
+			expectedError: accumPackage.NoPositionError{Name: invalidPositionName},
+		},
```

Comment:
would it be possible to add test case for position num shares being zero and having it deleted?

Reply:
I'm unsure what you mean by this.

Assuming that a test case where the rewards added are 0 or empty.

Added here: [ec2754d](https://github.com/osmosis-labs/osmosis/pull/5057/commits/ec2754d64fec91a1cb8c7d2fb128b75fda6f276e)